appinventor/components/src/com/google/appinventor/components/runtime/util/FroyoUtil.java

```diff
@@ -129,6 +129,7 @@ public static WebViewClient getWebViewClient(final boolean ignoreErrors,
     return new WebViewClient() {
       @Override
       public boolean shouldOverrideUrlLoading(WebView view, String url) {
+        EventDispatcher.dispatchEvent(this, "BeforePageLoad", url);
```

Comment:
Use `component` instead of `this`. `this` in this context will refer to the WebViewClient object, not the Web component (which is what it needs to be).

Reply:
Oops! Fixed.